x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/suggestion_panel.tsx

```diff
@@ -117,70 +138,241 @@ function InnerSuggestionPanel({
+              onClick={() => {
+                dispatch({
+                  type: 'SUBMIT_SUGGESTION',
+                });
+              }}
+            >
+              {i18n.translate('xpack.lens.sugegstion.confirmSuggestionLabel', {
+                defaultMessage: 'Confirm and reload suggestions',
+              })}
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroup>
+
       <div className="lnsSuggestionsPanel__suggestions">
-        {suggestions.map((suggestion: Suggestion) => (
+        {currentStateExpression && (
           <SuggestionPreview
-            suggestion={suggestion}
-            dispatch={dispatch}
-            frame={frame}
+            preview={{
+              expression: currentStateExpression
+                ? prependKibanaContext(currentStateExpression, expressionContext)
+                : undefined,
+              icon: 'empty',
+              title: i18n.translate('xpack.lens.suggestions.currentVisLabel', {
+                defaultMessage: 'Current',
+              }),
+            }}
             ExpressionRenderer={ExpressionRendererComponent}
-            previewExpression={
-              suggestion.previewExpression
-                ? preparePreviewExpression(
-                    suggestion.previewExpression,
-                    datasourceMap,
-                    datasourceStates,
-                    frame,
-                    suggestion.datasourceId,
-                    suggestion.datasourceState
-                  )
-                : undefined
-            }
-            key={`${suggestion.visualizationId}-${suggestion.title}`}
+            onSelect={rollbackToCurrentVisualization}
+            selected={lastSelectedSuggestion === -1}
+            showTitleAsLabel
           />
-        ))}
+        )}
+        {suggestions.map((suggestion, index) => {
+          return (
+            <SuggestionPreview
+              preview={{
+                expression: suggestion.previewExpression
+                  ? prependKibanaContext(suggestion.previewExpression, expressionContext)
+                  : undefined,
+                icon: suggestion.previewIcon,
+                title: suggestion.title,
+              }}
+              ExpressionRenderer={ExpressionRendererComponent}
+              key={index}
+              onSelect={() => {
+                if (lastSelectedSuggestion === index) {
+                  rollbackToCurrentVisualization();
+                } else {
+                  setLastSelectedSuggestion(index);
+                  switchToSuggestion(frame, dispatch, suggestion);
+                }
+              }}
+              selected={index === lastSelectedSuggestion}
+            />
+          );
+        })}
       </div>
     </div>
   );
 }
 
+interface VisualizableState {
+  visualizationState: unknown;
+  datasourceState?: unknown;
+  datasourceId?: string;
+  keptLayerIds?: string[];
+}
+
+function getPreviewExpression(
```

Comment:
This feels like something that should be in a `selector` module of some kind, rather than in a React component file. But not a big deal to keep it here, either.